tests/grouping/test_pauli_group.py

```diff
@@ -196,3 +202,58 @@ def test_pauli_mult_with_phase(self, pauli_word_1, pauli_word_2, wire_map, expec
         """Test that multiplication including phases works as expected."""
         _, obtained_phase = pauli_mult_with_phase(pauli_word_1, pauli_word_2, wire_map=wire_map)
         assert obtained_phase == expected_phase
+
+
+class TestPartitionPauliGroup:
+    """Tests for the partition_pauli_group function"""
+
+    def test_expected_answer(self):
+        """Test if we get the expected answer for 3 qubits"""
+        expected = [
+            ["III", "IIZ", "IZI", "IZZ", "ZII", "ZIZ", "ZZI", "ZZZ"],
+            ["IIX", "IZX", "ZIX", "ZZX"],
+            ["IIY", "IZY", "ZIY", "ZZY"],
+            ["IXI", "IXZ", "ZXI", "ZXZ"],
+            ["IXX", "ZXX"],
+            ["IXY", "ZXY"],
+            ["IYI", "IYZ", "ZYI", "ZYZ"],
+            ["IYX", "ZYX"],
+            ["IYY", "ZYY"],
+            ["XII", "XIZ", "XZI", "XZZ"],
+            ["XIX", "XZX"],
+            ["XIY", "XZY"],
+            ["XXI", "XXZ"],
+            ["XXX"],
+            ["XXY"],
+            ["XYI", "XYZ"],
+            ["XYX"],
+            ["XYY"],
+            ["YII", "YIZ", "YZI", "YZZ"],
+            ["YIX", "YZX"],
+            ["YIY", "YZY"],
+            ["YXI", "YXZ"],
+            ["YXX"],
+            ["YXY"],
+            ["YYI", "YYZ"],
+            ["YYX"],
+            ["YYY"],
+        ]
+        assert expected == partition_pauli_group(3)
+
+    @pytest.mark.parametrize("n", range(1, 9))
```

Comment:
for the case of `partition_pauli_group(0)` I get `[['']]` but maybe an exception would be appropriate?

Reply:
Added some validation here: 7b804bd52818c73bcca60009ec0eace02066f277